src/app/common/directives/oc-carousel/oc-carousel.js

```diff
@@ -0,0 +1,35 @@
+angular.module('orderCloud')
+    .directive('ocCarousel', ocCarouselDirective)
+;
+
+function ocCarouselDirective($compile, $templateRequest, $exceptionHandler) {
+    return {
+        scope: {
+            buyer: '=',
+            slides: '='
+        },
+        restrict: 'E',
+        link: function(scope, element) {
+            if(typeof scope.buyer === 'undefined'){
+                $exceptionHandler({message: 'ocCarousel directive is not configured correctly, missing buyer object data'});
+            }
+            if((!scope.buyer) && typeof scope.slides === 'undefined'){
+                $exceptionHandler({message: 'ocCarousel directive is not configured correctly, missing slides data'});
+            }
```

Comment:
You're not handling the scenario where buyer is defined, but the slides property on buyer is not. In which case there would still not be anything displaying.

You could define a shorthand handler for slides and at the same time verify its existence. Maybe something like:
`var slides = (buyer.xp && buyer.xp.Slides && buyer.xp.Slides.Items && buyer.xp.Slides.Items.length) ? buyer.xp.Slides : null`

then use that for your $exceptionHandler and then below for your other conditional